src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpGoogleCloudStorageDAO.scala

```diff
@@ -56,76 +57,35 @@ class HttpGoogleCloudStorageDAO(
-    val bucketCredential = getBucketCredential(userInfo)
-    // ACLs are stored internally in three Google groups.
-    groupAccessLevelsAscending foreach { level => createGoogleGroupWithACL(bucketCredential, level, bucketName, workspaceName) }
-
-    // add the owner -- may take up to 1 minute after groups are created
-    // (see https://developers.google.com/admin-sdk/directory/v1/guides/manage-groups)
-    val targetMember = new Member().setEmail(userInfo.userEmail).setRole(groupMemberRole)
-    val directory = getGroupDirectory(getGroupServiceAccountCredential)
-    directory.members.insert(makeGroupId(workspaceName, WorkspaceAccessLevel.Owner), targetMember).execute()
-  }
-
-  override def teardownACL(bucketName: String, workspaceName: WorkspaceName): Unit = {
-    groupAccessLevelsAscending foreach { level => deleteGoogleGroup(level, bucketName, workspaceName) }
-  }
-
-  def createGoogleGroupWithACL(bucketCredential: Credential, accessLevel: WorkspaceAccessLevel, bucketName: String, workspaceName: WorkspaceName): Unit = {
-    val groupId = makeGroupId(workspaceName, accessLevel)
-    val group = new Group().setEmail(groupId).setName(s"FireCloud ${workspaceName.namespace}/${workspaceName.name} ${accessLevel}")
-
-    // create group
-    val serviceCredential = getGroupServiceAccountCredential
-    val directory = getGroupDirectory(serviceCredential)
-    directory.groups().insert(group).execute()
-
-    // add group to ACL (requires bucket owner's credential)
-    val storage = getStorage(bucketCredential)
-
-    val bucketAcl = new BucketAccessControl().setEntity(makeGroupEntityString(groupId)).setRole(WorkspaceAccessLevel.toCanonicalString(accessLevel)).setBucket(bucketName)
-    storage.bucketAccessControls().insert(bucketName, bucketAcl).execute()
-
-//    val objectAcl = new ObjectAccessControl().setEntity(makeGroupEntityString(groupId)).setRole(WorkspaceAccessLevel.toCanonicalString(accessLevel)).setBucket(bucketName)
-//    storage.defaultObjectAccessControls().insert(bucketName, objectAcl).execute()
-  }
-
-  def deleteGoogleGroup(accessLevel: WorkspaceAccessLevel, bucketName: String, workspaceName: WorkspaceName): Unit = {
-    val groupId = makeGroupId(workspaceName, accessLevel)
-    val serviceCredential = getGroupServiceAccountCredential
-    val directory = getGroupDirectory(serviceCredential)
-    directory.groups.delete(groupId).execute()
-  }
-
-  override def createBucket(userInfo: UserInfo, projectId: String, bucketName: String): Unit = {
-    val credential = getBucketCredential(userInfo)
-    val storage = getStorage(credential)
-    val bucket =  new Bucket().setName(bucketName)
-    storage.buckets().insert(projectId, bucket).execute()
+  override def createBucket(userInfo: UserInfo, projectId: String): Try[String] = {
+    val buckets = getStorage(getBucketCredential(userInfo)).buckets
+    retry(2) { // in case of bucket name collisions
+      val workspaceId = UUID.randomUUID.toString
```

Comment:
it is unexpected to create the workspace id within the createBucket function. Make workspaceId a parameter.